addons/account/models/account_move.py

```diff
@@ -2045,17 +2058,17 @@ def dirty():
         if needed_after == needed_before:
             return
 
-        to_delete = sorted({
+        to_delete = [
             line.id
             for key, line in existing_before.items()
             if key not in needed_after
             and key in existing_after
             and before2after[key] not in needed_after
-        } | {
-            line.id
+        ]
+        to_delete.extend( line.id
             for key, line in existing_after.items()
-            if key not in needed_after
-        })
+            if key not in needed_after and line.id not in to_delete
```

Comment:
I'm trying to keep this in O(n), could you save a copy of `to_delete` as a set to have the `in` operation in O(1)